schema/chat_room.proto

```diff
@@ -0,0 +1,288 @@
+
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/longrunning/operations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/error_details.proto";
+
+package google.showcase.v1alpha3;
+
+option go_package = "github.com/googleapis/gapic-showcase/server/genproto";
+option java_package = "com.google.showcase.v1alpha3";
+option java_multiple_files = true;
+
+option (google.api.metadata) = {
+  product_uri: "https://github.com/googleapis/gapic-showcase"
+};
+
+service ChatRoom {
+  rpc CreateRoom(CreateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["display_name", "description"]
+    };
+  }
+
+  rpc GetRoom(GetRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc UpdateRoom(UpdateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{room.name=rooms/*}"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["room.name", "room.display_name"]
+    };
+    option (google.api.method_signature) = {
+        fields: ["room.name", "room.display_name", "room.description"]
+    };
+  }
+
+  rpc DeleteRoom(DeleteRoomRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  rpc CreateMessage(CreateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}/messages"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent", "message"]
+    };
+  }
+
+  rpc GetMessage(GetMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*/messages/*}"
+    };
+  }
+
+  rpc UpdateMessage(UpdateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{message.name=rooms/*/messages/*}"
+      body: "*"
+    };
+  }
+
+  rpc DeleteMessage(DeleteMessageRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*/messages/*}"
+    };
+  }
+
+  rpc ListMessages(ListMessagesRequest) returns (ListMessagesResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/messages"
+    };
+  }
+
+  rpc SearchMessages(SearchMessagesRequest) returns (google.longrunning.Operation) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms/-/messages:search"
```

Comment:
Would it make sense to _allow_ searching within a particular room? (e.g. `"/v1alpha3/{parent=rooms/*}/messages:search"`)

Reply:
That definitely makes sense, but I think that making the search space smaller makes it less convincing that this is a LRO.

Reply:
But it is an LRO if _any_ request that comes from it would take time.
(Also, have you ever searched messages in Slack when there is a multi-year history. Fear not, it is totally reasonable for this to be an LRO ;-))